netrd/distance/graph_diffusion_distance.py

```diff
@@ -0,0 +1,156 @@
+2013 IEEE (pp 419-422). IEEE. https://doi.org/10.1109/GlobalSIP.2013.6736904
+
+author: Brennan Klein
+email: brennanjamesklein at gmail dot com
+Submitted as part of the 2019 NetSI Collabathon.
+
+"""
+
+import numpy as np
+import networkx as nx
+from scipy.sparse.csgraph import laplacian
+from .base import BaseDistance
+from ..utilities import undirected
+
+
+class GraphDiffusion(BaseDistance):
+    """Find the maximally dissimilar diffusion kernels between two graphs."""
+
+    @undirected
+    def dist(self, G1, G2, thresh=1e-14, resolution=1000):
+        r"""The graph diffusion distance between two graphs, :math:`G` and :math:`G'`,
+        is a distance measure based on the notion of flow within each graph. As
+        such, this measure uses the unnormalized Laplacian matrices of both
+        graphs, :math:`\mathcal{L}` and :math:`\mathcal{L}'`, and uses them to
+        construct time-varying Laplacian exponential diffusion kernels,
+        :math:`e^{-t\mathcal{L}}` and :math:`e^{-t\mathcal{L}'}`, by
+        effectively simulating a diffusion process for :math:`t` timesteps,
+        creating a column vector of node-level activity at each timestep. The
+        distance :math:`d_\texttt{GDD}(G, G')` is defined as the Frobenius norm
+        between the two diffusion kernels at the timestep :math:`t^{*}` where
+        the two kernels are maximally different.
+
+        .. math::
+            D_{GDD}(G,G') = \sqrt{||e^{-t^{*}\mathcal{L}}-e^{-t^{*}\mathcal{L}'}||}
+
+        The results dictionary also stores a 2-tuple of the underlying
+        adjacency matrices in `adjacency_matrices`, the Laplacian matrices in
+        `laplacian matrices`, and the output of the optimization process
+        (`peak_diffusion_time` and `peak_deviation`).
+
+        Adapted from the authors' MATLAB code, available at: https://rb.gy/txbfrh
+
+
+        Parameters
+        ----------
+
+        G1, G2 (nx.Graph)
+            two networkx graphs to be compared.
+
+        thresh (float)
+            minimum value above which the eigenvalues will be considered.
+
+        resolution (int)
+            number of :math:`t` values to span through.
+
+        Returns
+        -------
+        dist (float)
+            the distance between `G1` and `G2`.
+
+        References
+        ----------
+
+        .. [1] Hammond, D. K., Gur, Y., & Johnson, C. R. (2013, December).
+               Graph diffusion distance: A difference measure for weighted graphs based on the
+               graph Laplacian exponential kernel. In Global Conference on Signal and
+               Information Processing, 2013 IEEE (pp 419-422). IEEE.
+               https://doi.org/10.1109/GlobalSIP.2013.6736904
+
+        """
+
+        A1 = nx.to_numpy_array(G1)
+        A2 = nx.to_numpy_array(G2)
+
+        L1 = laplacian(A1)
+        L2 = laplacian(A2)
+
+        D1, V1 = np.linalg.eig(L1)
+        D2, V2 = np.linalg.eig(L2)
```

Comment:
Suggestion: I use `vals, vecs = np.eig()` always. For this I suggest `vals1, vecs2` and `vals2, vecs2`.

Reply:
Done. This is a common thing to do in the library, we should make sure our names are consistent (I'm sure they aren't.